[Teams][CreatedTogetherModedocument][1477618] (#2710)
MicrosoftDocs/msteams-docs: msteams-platform/apps-in-teams-meetings/teams-together-mode.md

---
title: Work in Together Mode in Teams
description: Work with Together Mode 
ms.topic: conceptual
---

> # Work in Together Mode in Teams

Author: Already changed

---
title: Together Mode in Teams
description: Work with Together Mode 
ms.topic: conceptual
---

# Together Mode in Teams

> Teams Together Mode provides a simple, innovative, and user-friendly meeting environment. It digitally combines participants into a single virtual space instead of being shown separately in boxes. It creates your virtual image and places the image in a virtual environment where you can easily interact with your peers. It makes meetings more engaging in the auditorium view.

Reviewer: Please reword this paragraph: 

Microsoft Teams Together Mode provides an immersive and engaging meeting environment that brings people together and encourages them to turn on their video. It digitally combines participants into a single virtual scene and places their video streams in pre-determined spots designed and fixed by the scene creator.

> [!VIDEO https://www.youtube-nocookie.com/embed/MGsNmYKgeTA]

You can create scene only apps or extend your current app to have scenes for Together Mode. A scene is a virtual view of the auditorium or meeting venue where the attendees are placed and includes the seats for attendees.

> 

Reviewer: A scene is more than auditorium. It can be anything that the scene developer wants. Lets reword this: 

**Scenes** in Together Mode is an artifact created by the scene developer using the _scene design studio_  (we will need another page where we will explain an end to end scene building exercise in scene design studio) that brings together people along-with their video stream in a creative setting as conceived by the scene creator. A scene brings people together by having designated **seats** in the conceived scene setting and participant video streams are rendered in those seats.

Author: Please provide the details as per this comment "we will need another page where we will explain an end to end scene building exercise in scene design studio"

>[!NOTE]

> > Scene only apps are recommended as the acquisition experience for such apps is more seamless.

Reviewer: Add this statement: 

A scene only app is still an app in Microsoft Teams. The app package creation step is abstracted out since the scene design studio handles the app package creation in the background.

Author: Added the statement as a note as the statement is more of additional information.

> The following image gives an overview of adding a scene to an app:

Reviewer: Reword: 
The following image gives an overview to create a scene only app.

> ![Add scene to app](../assets/images/apps-in-meetings/scene-added-to-app.png)

Reviewer: Why do I see a border rectangle? Anyway, can we please use a better way to represent this? I used SmartArt in powerpoint to help drive the point but can you use some way to represent this that looks good in our docs? 

Also, the steps have changed a bit:

1. Build your scene with the Microsoft Scene Design Studio 
2. Give your scene a name and Save 
3. Test your scene in Microsoft Teams 
4. Ship your scene to Microsoft Teams or Use it just for yourself or create a link instantly and share your scene without shipping it as a third-party scene to Microsoft Teams

Author: We need to add black borders to all images as per standard followed.

Author: Please suggest a better way to show the image as images are required to be as per Microsoft standard and changing the color or style will probably not follow standard.

Reviewer 2: @v-rodriguezd Image borders are enabled by default but optional. Removing borders from diagrams like this I think makes sense from a design perspective. You can do this by adding `border="false"` to your image Markdown like this:

`:::image type="content" source="<folderPath>" alt-text="<alt text>" border="false":::`

>[!NOTE]

> > Multiple scenes in a single app package appear as a flat list of scenes to users.

Reviewer: Add a picture here. I will give you one

Author: @rarang please provide the image.

There are prerequisites that you must complete before using the Together Mode.

## Prerequisites

> You must have a basic understanding of the following to use Together Mode:

Reviewer: I think the pre-reqs here are very vague. Please replace with this following: 

1. Definition of Scene and Seats in a scene 
2. Have a Microsoft Developer account and familiarity with the Microsoft Teams Dev Center and App Studio
3. Concept of app sideloading (insert a link here to this page : https://review.docs.microsoft.com/en-us/microsoftteams/platform/concepts/deploy-and-publish/apps-upload?branch=pr-en-us-2710

* Scene
* Scene Design Studio
* Create apps in App Studio
* Sideload or upload apps in Teams

> Now you can create or build a scene in the Scene Design Studio using Scene editor, where you can add meeting participants, add background images, and align images to the scene.

Reviewer: You don't need this line

## Build a scene using the Scene Design Studio

> 

Reviewer: Add a note here: 

This document is referring to scene design studio in the Microsoft Teams Dev Center. The interface and functionalities are all the same in App Studio Scene Designer.

> Microsoft has a Scene Design Studio that allows you to build the scenes. It is available on the [Teams Dev Center](https://dev.teams.microsoft.com/home).

Reviewer: Small correction - (https://dev.teams.microsoft.com/**scenes**)

> Together Mode allows you to apply different scenes while conducting meetings in a virtual environment.

Reviewer: You don't need this line. Please remove it

> A scene is an artifact that contains the following:

Reviewer: Reword: 
A scene in the context of the scene design studio is an artifact that contains the following

> * Seats of meeting participants if their videos are turned on.

Reviewer: Re-order and Re-word as follows: 

1. Seats reserved for Meeting Organizer and Meeting Presenters. Note that Presenters here does not refer to the user who actively sharing. It refers to the [meeting role](https://support.microsoft.com/en-us/office/roles-in-a-teams-meeting-c16fa7d0-1666-4dde-8686-0a0bfe16e019)
2. Each seat and image have a width and height. Note that PNG is the only supported format
3. The xyz coordinates of all the seats and images

* Each seat is placed on an XY canvas.
* A collection of images that are camouflaged as one image.
* Each seat and image has width and height.
* Optionally, you can reserve a spot for the meeting organizer and a set of reserved spots for participants.

![Scene Design Studio](../assets/images/apps-in-meetings/scene-design-studio.png)

> In the Scene Design Studio, each seat represents an avatar for design purposes. Each seat and image has a placement in the third dimension, that is the Z-axis to give a sense of depth in scenes that require them.

Reviewer: Reword: 

As depicted in the above image, each seat is represented as an avatar for development purposes. The seat dimensions here ultimately before the canvas for rendering the participant video stream.

Author: The import scene statement have added after the Build a scene steps as it looks like an additional step that can be performed.

![Build a scene](../assets/images/apps-in-meetings/build-a-scene.png)

**To build a scene using the Scene Design Studio**

> 1. Go to [Teams Dev Center](https://dev.teams.microsoft.com/home).

Reviewer: (https://dev.teams.microsoft.com/**scenes**)

> 2. Select **Tools** from the left-hand section and then select **Scene editor** under **Tools** to open the Scene Design Studio.

Reviewer: Reword: 

Select **Tools** from the menu and then select **Scene editor** under **Tools** to open the Scene Design Studio.

3. From the **Scenes Editor** page, select **Create a new scene**.

> 

Reviewer: Add step: Name your scene on the right side

4. Select **Participants** under **Layers** in the upper-right corner.

5. Select the number of participants for the scene from the **Number of participants** box, and select **Add**. After the scene is shipped, the avatar placements are replaced with actual participant's video streams.

> 6. Drag and drop the image into the environment as displayed in the following image:

Reviewer: This step should come before step 4

Reviewer: I @ mentioned Deborah in a thread. Can we get those assets attached here so that anybody trying this out can simply download the asset and can try it out themselves as they read through the doc.

Author: Please suggest how to add the assets is it by creating a folder in apps-in-teams-meetings and attaching Sample scene1 and scene2 zip files.

Author: Added in step 5 as note please check

Author: Can we rename the assets to Sample Scene.zip and Sample App.zip.

    >[!NOTE]
    > Alternately, you can add background images to the scene using **Add images**.

    ![Drag into the scene](../assets/images/apps-in-meetings/drag-and-drop-scene.png)

7. Select an image that you have placed in the scene. From the rightmost pane, select an alignment for the image or use the **Size & rotation** slider to adjust the image size.

    >[!NOTE]
    > Alternately, you can drag the images around the scene and place them in the required position and resize them using the resize arrow.

    ![Alignment for images](../assets/images/apps-in-meetings/image-alignment.png)

8. Select any participant image, select the **Assign Spot** check box, and select **Meeting Organizer** and **Presenter** to assign that spot to the participant.

    >[!NOTE]
    > Alternately, you can select either **Meeting Organizer** or **Presenter** options for that participant.

    ![Assign spot](../assets/images/apps-in-meetings/assign-spot.png)

> 9. In the **Scene Name** box, enter a name for the scene.

Author: This is not showing as step 9. Please check the document and let me know.

> 10. Select **Export the Scene** from the **Save** drop-down menu. A .ZIP file, that is the scene package, is downloaded.

Author: Select Preview step is already there in step 9. Somehow your review shows older version of document.

    >[!NOTE]
    > Alternately, you can select **Save** to save the scene.

After you have built a scene using the Scene Design Studio, you can assemble a scene package.

> ## Assemble a scene package

Author: The Assemble a scene package section is not there.

         "height":4.0,
         "zOrder":2.0,
   "isAlpha":true
      }

   ],
   "participants":[
      {
         "cx":5.0,
         "cy":4.0,
         "width":4.0,
         "height":2.25,
         "zOrder":1.0,
         "seatingOrder":0
      },
      {
         "cx":11.0,
         "cy":4.0,
         "width":4.0,
         "height":2.25,
         "zOrder":1.0,
         "seatingOrder":1
      }
   ]
}
```

Each scene has a unique ID and name. The scene JSON also contains information on all the assets used for the scene. Each asset contains a filename, width, height, and position on the X and Y-axis. Similarly, each seat contains a seat ID, width, height, and position on the X and Y-axis. The seating order is generated automatically for each seat when it is created.

>[!NOTE]
> Seating order number corresponds to the order of people joining the call.

The seating order can be altered according to the preferred order.

The zOrder represents the order of placing images and seats along the Z-axis. In many cases, it gives a sense of depth or partition if required.

Now that you have assembled a scene package, you can create an app package and apply the scene in Teams.

> ## Create an app package and use the scene in Teams

Author: This section is not there.

An app package contains the scene package. App Studio simplifies app packaging.

![App Studio](../assets/images/apps-in-meetings/app-studio.png)

**To create an app package and use the scene in Teams**

1. Go to **Apps** in **App Studio** in Teams.

2. Create a new app and enter the necessary details in the app details page.

3. In the **Microsoft Teams Developer Center** page, go to **Scenes** under **Configuration** in **Apps**.

4. Select the scene you built and import the package.

5. Sideload this package in Teams.

Now that you have created an app package and uploaded the scene in Teams, you can activate the Together Mode to engage in scenes.

## Activate the Together Mode

Get end-to-end information of how an end user engages with scenes in Together Mode.

**To choose scenes and activate the Together Mode**

1. From the **Gallery** drop-down in the upper-left corner, select **Change scene** to change the default scene. The **Choose a scene** dialog box appears.

2. From **Choose a scene**, select the scene you want to use for your meeting.

>     ![Scene picker](../assets/images/apps-in-meetings/scene-picker.png)

Author: This image is not there.

* Seat and image for each participant. Each seat and image has an adjustable width and height.

    >[!NOTE]
    > PNG is the only supported format.

* The xyz coordinates of all the seats and images.
* A collection of images that are camouflaged as one image.

The following image shows each seat represented as an avatar for building scenes:

![Scene studio](../assets/images/apps-in-meetings/scene-design-studio.png)

The seat dimensions become the canvas for rendering the participant video stream.

![Build a scene](../assets/images/apps-in-meetings/build-a-scene.png)

**To build a scene using the Scene studio**

1. Go to [Scenes Editor - Teams Developer Portal](https://dev.teams.microsoft.com/scenes).
 
    >[!NOTE]
    > * To open Scene studio, you can navigate to the home page of [Teams Developer Portal](https://dev.teams.microsoft.com/home) and select **Scene studio** from the **Create custom scenes for meetings** section.
    > * To open Scene studio, you can navigate to the home page of [Teams Developer Portal](https://dev.teams.microsoft.com/home), select **Tools** from the left hand section, and select **Scene studio** from the **Tools** section.

> 2. From the **Scenes Editor** page, select **Create a new scene**.

Reviewer 3: ```sugestion

1. In the **Scenes Editor** page, select **Create a new scene**
```

> 3. On the right side, in the **Scene Name** box, enter a name for the scene.

Reviewer 3: ```suggestion
1. In the **Scene Name** box, enter a name for the scene.
```

    >[!NOTE]
    > * You can select **Close** to toggle between closing or reopening the right pane.
    > * You can zoom in or zoom out of the scene using the zoom bar for a better view of the scene.

> 4. Drag and drop the image into the environment as displayed in the following image:

Reviewer 3: ```suggestion
1. Drag and drop the image into the environment as displayed in the following image:
```

    >[!NOTE]
    > * You can download the [SampleScene.zip](/apps-in-teams-meetings/SampleScene.zip) and [SampleApp.zip](/apps-in-teams-meetings/SampleApp.zip) files with the images.
    > * Alternately, you can add background images to the scene using **Add images**.

    ![Drag into the scene](../assets/images/apps-in-meetings/drag-and-drop-scene.png)

> 5. Select an image that you have placed in the scene. From the right pane, select an alignment for the image or use the **Resize** slider to adjust the image size.

Reviewer 3: ```suggestion
1. Select an image that you have placed in the scene.
1. From the right pane, select an alignment for the image or use the **Resize** slider to adjust the image size.
```

    ![Alignment for images](../assets/images/apps-in-meetings/image-alignment.png)

> 6. Select **Participants** under **Layers** in the upper-right corner.

Reviewer 3: ```suggestion
1. In the upper-right corner, select **Participants** under **Layers**.
```

> 7. Select the number of participants for the scene from the **Number of participants** box, and select **Add**.

Reviewer 3: All numbers in Markdown should only use "1...." you need not write 1, 2, 3

    >[!NOTE]
    > * After the scene is shipped, the avatar placements are replaced with actual participant's video streams.
    > * You can drag the participant images around the scene and place them in the required position and resize them using the resize arrow.

> 8. Select any participant image, and choose the **Assign Spot** check box to assign the spot to the participant.

Reviewer 3: Same comment as above for all the numbers.

         "cy":4.0,
         "width":4.0,
         "height":2.25,
         "zOrder":1.0,
         "seatingOrder":1
      }
   ]
}
```

Each scene has a unique ID and name. The scene JSON also contains information on all the assets used for the scene. Each asset contains a filename, width, height, and position on the X and Y-axis. Similarly, each seat contains a seat ID, width, height, and position on the X and Y-axis. The seating order is generated automatically for each seat when it is created.

>[!NOTE]
> Seating order number corresponds to the order of people joining the call.

The seating order can be altered according to the preferred order.

The zOrder represents the order of placing images and seats along the Z-axis. In many cases, it gives a sense of depth or partition if required. For more information, see the step-by-step getting started sample. The sample leverages the zOrder.

Now that you have gone through the sample scene.json, you can activate the Together Mode to engage in scenes.

## Activate the Together Mode

Get end-to-end information of how an end user engages with scenes in Together Mode.

**To choose scenes and activate the Together Mode**

> 1. Create a new test meeting.

Reviewer 3: Correct Markdown syntax for numbering. See the comments above.
